### PR TITLE
Adding new routes from the js stuff

### PR DESCRIPTION
**Pull request by perryprog**
I lied I'm making the PR today.

The below is copied from the initial commit message, sorry for the sloppiness.

---
I added a Endpoint.setup thing that does a lot of the repetitive work for you, it will probably be expanded. When using it you need to deconstruct the result in a form of ((avatar1,...,) base).

The base should be the name of the image and dir, with an optional extension. I may change this to automatically grab the images and save it in a cache on startup though. Idk, I'll figure something out. The point is the directory needs to be the same as the omage name so some of those will be changed.

I'm planning on moving what I can to the new self.setup style, but I doubt it'll work for all routes with no issues.

---

### New routes to finish

- [x] Brazzers
- [x] cancer
- [x] cry
- [x] dab
- [x] delete
- [x] door
- [x] egg
- [x] failure
- [x] fakenews
- [x] fedora
- [x] floor
- [x] laid
- [x] note
- [x] ohno
- [x] plan
- [x] rip
- [x] satan
- [x] savehumanity
- [x] thesearch 

`dank.js` and `salty.js` will be ported later as it's super involved and hard. (they're gifs)

---
### ~~Routes to change to this.setup. ~~

It was decided that decisions about this should be done in a separate PR. Routes will be made in the old style for now.

~~Some of the below will not gain anything from using this, and will be moved into a separate category as I get to them.~~
- [x] brazzers
- [x] slap
- [x] abandon
- [x] bed
- [x] brain
- [x] byemom
- [x] deepfry
- [x] disability
- [x] facts             
- [x] gay
- [x] hitler
- [x] invert
- [x] jail
- [x] magik
- [x] quote
- [x] shit
- [x] sickfilth
- [x] spank
- [x] trash
- [x] trigger
- [x] tweet
- [x] ugly
- [x] warp
- [x] whodidthis

### CODE REVIEW

File: endpoints/dab.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+from utils import http
+from utils.endpoint import Endpoint
+
+
+class Dab(Endpoint):
+    def generate(self, avatars, text, usernames):
+        base = Image.open('assets/dab/dab.png').convert('RGBA')
+        avatar = Image.open(http.get_image(avatars[0])).resize((500, 500)).convert('RGBA')
+        final_image = Image.new('RGBA', base.size)
```
Comment (devoxin) [inline endpoints/dab.py:14]: Why not just use the avatar as the base?

File: endpoints/floor.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+from utils.endpoint import Endpoint
+from utils.textutils import wrap
+
+
+class Floor(Endpoint):
+    def generate(self, avatars, text, usernames):
+        base = Image.open('assets/floor/floor.jpg').convert('RGBA')
+        # Open it twice because python
+        avatar = Image.open(http.get_image(avatars[0])).resize((45, 45)).convert('RGBA')
+        avatar2 = Image.open(http.get_image(avatars[0])).resize((23, 23)).convert('RGBA')
```
Comment (devoxin) [inline endpoints/floor.py]: Use .copy() here to reduce fetch requests

**Review by devoxin — CHANGES_REQUESTED** [review]

**Comment by devoxin** [comment]
On another note why do you make a new layer to be used as the textlayer? Would it not be faster to use the base image with ImageDraw?

**Comment by perryprog** [comment]
Probably because it was done like that in other text images, and I was lazy and just copy pasted. I'll switch it over to drawing on the base image.

**Comment by perryprog** [comment]
If the text is rotated then it seems like the best way to rotate it is with a separate text_layer.

File: endpoints/floor.py [hunk 3]
```diff
@@ -11,19 +11,16 @@
 class Floor(Endpoint):
     def generate(self, avatars, text, usernames):
         base = Image.open('assets/floor/floor.jpg').convert('RGBA')
-        # Open it twice because python
         avatar = Image.open(http.get_image(avatars[0])).resize((45, 45)).convert('RGBA')
-        avatar2 = Image.open(http.get_image(avatars[0])).resize((23, 23)).convert('RGBA')
-        text_layer = Image.new('RGBA', base.size)
+        avatar2 = avatar.copy()
```
Comment (devoxin) [inline endpoints/floor.py]: Did you forget to resize to 23x23 lol

Comment (perryprog) [reply endpoints/floor.py]: oh no

**Review by devoxin — APPROVED** [review]

**Pull request merged**